tools/nxfuse/src/filemtd.c

```diff
@@ -161,17 +161,25 @@ static ssize_t filemtd_write(FAR struct file_dev_s *priv, size_t offset, FAR con
 	uint8_t srcvalue;
 	uint8_t newvalue;
 	size_t seekpos;
-
+	size_t bufsize = sizeof(buf);
+	size_t readsize = 0;
+	
 	/* Set the starting location in the file */
-
 	seekpos = priv->offset + offset;
-
-	while (len-- > 0) {
+	while (len > 0) {
 		if (buflen == 0) {
 			/* Read more data from the file */
 
 			lseek(priv->fd, seekpos, SEEK_SET);
-			buflen = read(priv->fd, buf, sizeof(buf));
+			if (len < bufsize) {
+				buflen = len;
+			} else {
+				buflen = bufsize;
+			}
+			readsize = read(priv->fd, buf, buflen);
+			if (readsize != buflen) {
+				return ERROR;
+			}
```

Comment:
Is len -= buflen required here?

Reply:
please understand role of this function, it is read specific size of data, and check bit byte by byte. len is decreased at the end of loop.